python_ta/checkers/possibly_undefined_checker.py

```diff
@@ -0,0 +1,119 @@
+           }
+
+    # this is important so that your checker is executed before others
+    priority = -1
+
+    def __init__(self, linter=None):
+        super().__init__(linter=linter)
+        self._possibly_undefined: Set[astroid.Name] = set()
+
+    @check_messages('possibly-undefined')
+    def visit_name(self, node):
+        """Adds message if there exists a path from the start block to node where
+        a variable used by node might not be defined."""
+        if node in self._possibly_undefined:
+            self.add_message('possibly-undefined', node=node)
+
+    def visit_module(self, node):
+        self._analyze(node)
+
+    def visit_functiondef(self, node):
+        self._analyze(node)
+
+    def _analyze(self, node: Union[astroid.Module, astroid.FunctionDef]) -> None:
+        """Runs the data flow algorithm on a `Module` or `Function` CFG, which in turn
+        appends `Name` nodes to `self.possibly_undefined` if it might not be defined.
+
+        Data flow algorithms retrieved from:
+        https://www.seas.harvard.edu/courses/cs252/2011sp/slides/Lec02-Dataflow.pdf#page=31
+        """
+        out_facts = {}
+        cfg = ControlFlowGraph()
+        cfg.start = node.cfg_block
+        blocks = list(cfg.get_blocks_postorder())
+        blocks.reverse()
+
+        all_assigns = self._get_assigns(node)
+        for block in blocks:
+            out_facts[block] = all_assigns.copy()
+
+        worklist = blocks
+        while len(worklist) != 0:
+            b = worklist.pop()
+            outs = [out_facts[p.source] for p in b.predecessors if p.source in out_facts]
+            if outs == []:
+                in_facts = set()
+            else:
+                in_facts = set.intersection(*outs)
+            temp = self._transfer(b, in_facts, all_assigns)
+            if temp != out_facts[b]:
+                out_facts[b] = temp
+                worklist.extend([succ.target for succ in b.successors])
+
+    def _transfer(self, block: CFGBlock, in_facts: Set[str], local_vars: Set[str]) -> Set[str]:
+        gen = in_facts.copy()
+        kill = set()
+        for statement in block.statements:
+            if isinstance(statement, astroid.FunctionDef):
+                continue
+            for node in statement.nodes_of_class((astroid.AssignName, astroid.DelName, astroid.Name),
+                                              astroid.FunctionDef):
+                if isinstance(node, astroid.AssignName):
+                    gen.add(node.name)
+                elif isinstance(node, astroid.DelName):
+                    kill.add(node.name)
+                else:
+                    name = node.name
+                    # comment out 'self.config....' check when running tests
+                    if not (name in astroid.Module.scope_attrs or utils.is_builtin(name) or
+                            name in self.config.additional_builtins ) \
```

Comment:
This specific check throws an attribute error:
<img width="1255" alt="Screen Shot 2019-09-22 at 3 22 01 PM" src="https://user-images.githubusercontent.com/18351436/65393283-dc9c6200-dd4c-11e9-999e-436850411e48.png">

Reply:
@kavin Got it, I realized what was missing. If you look at the [pylint code](https://github.com/PyCQA/pylint/blob/master/pylint/checkers/variables.py#L600), it includes a class variable `options` that contains a tuple specifying that the option "additional-builtins" is used. This is what initializes `self.config.additional_builtins`. Please add the same code to this checker too.

Reply:
I added the code pylint code but this caused another error to occur. I was able to try many different ideas to solve this issue but I'm currently at a dead end. 

<img width="1165" alt="Screen Shot 2019-09-28 at 7 17 33 PM" src="https://user-images.githubusercontent.com/18351436/65823459-abc49d00-e224-11e9-84ee-eb361f0017c7.png">

I'll look into it again in a couple of days.

Reply:
Okay I looked into this a bit more; looks like there's no easy way to reuse an option from an existing checker. You can remove the part that uses `additional_builtins` and the options!